packages/buidler-core/src/internal/solidity/dependencyGraph.ts

```diff
@@ -61,6 +61,66 @@ export class DependencyGraph {
     return [...transitiveDependencies];
   }
 
+  public getConnectedComponents(): DependencyGraph[] {
```

Comment:
This function is tested both with unit tests and through the compile task in some real projects. Still, I'm very unsure about this implementation.